lib/abide_dev_utils/cem/generate/reference.rb

```diff
@@ -341,17 +384,29 @@ def control_levels_builder
             return unless @control.levels
 
             @md.add_ul('Supported Levels:')
-            @control.levels.each do |l|
-              @md.add_ul(@md.code(l), indent: 1)
+            if @valid_level.empty?
+              @control.levels.each do |l|
+                @md.add_ul(@md.code(l), indent: 1)
+              end
+            else
+              @valid_level.each do |l|
+                @md.add_ul(@md.code(l), indent: 1)
+              end
```

Comment:
What's going on here? If the valid levels are empty we go through the control's levels and add it, but if the valid levels aren't empty we go through the valid levels and add an entry for each one? What purpose does this serve?

Reply:
My intention was if the valid levels are empty, that means that we didn't provide any levels or profiles to the generator so we go with all the levels and profiles by using `@control`

Reply:
The code could use a comment or two explaining that because it's not very intuitive. Wasn't the intent of these changes to remove this sort of checking at this level because we're checking twice?

Reply:
Yeah I moved the checking out of the functions `control_levels_builder` and `control_profiles_builder`. Now they just build after checking if we've provided them selected levels and profiles.